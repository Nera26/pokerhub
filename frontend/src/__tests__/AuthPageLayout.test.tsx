import { render } from '@testing-library/react';
import LoginPage from '@/features/login';
import ForgotPasswordPage from '@/features/forgot-password';

jest.mock('@/app/components/auth/LoginForm', () => () => <div>Login Form</div>);
jest.mock('@/app/components/auth/ForgotPasswordForm', () => () => <div>Forgot Password Form</div>);

jest.mock('next/navigation', () => ({
  useRouter: () => ({ push: jest.fn(), prefetch: jest.fn() }),
}));

describe('AuthPageLayout', () => {
  it('renders login page correctly', () => {
    const { container } = render(<LoginPage />);
    expect(container).toMatchSnapshot();
  });

  it('renders forgot password page correctly', () => {
    const { container } = render(<ForgotPasswordPage />);
    expect(container).toMatchSnapshot();
  });
});
