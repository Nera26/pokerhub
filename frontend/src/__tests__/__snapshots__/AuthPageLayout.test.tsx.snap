// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthPageLayout renders forgot password page correctly 1`] = `
<div>
  <main
    class="flex min-h-screen flex-col items-center justify-center gap-6"
    id="main-content"
  >
    <h1
      class="text-2xl font-bold"
    >
      Forgot Password
    </h1>
    <div>
      Forgot Password Form
    </div>
  </main>
</div>
`;

exports[`AuthPageLayout renders login page correctly 1`] = `
<div>
  <main
    class="flex min-h-screen flex-col items-center justify-center gap-6"
    id="main-content"
  >
    <h1
      class="text-2xl font-bold"
    >
      Login
    </h1>
    <div>
      Login Form
    </div>
    <a
      class="text-accent-yellow hover:underline"
      href="/forgot-password"
    >
      Forgot password?
    </a>
  </main>
</div>
`;
